diff --git b/Android.bp a/Android.bp
index d3d46248..bc1191a5 100644
--- b/Android.bp
+++ a/Android.bp
@@ -32,10 +32,10 @@ license {
     ],
 }
 
-rust_binary {
-    name: "crosvm",
+rust_defaults {
+    name: "crosvm_binary_defaults",
     defaults: ["crosvm_inner_defaults"],
-    host_supported: true,
+    prefer_rlib: true,
     crate_name: "crosvm",
     cargo_env_compat: true,
     cargo_pkg_version: "0.1.0",
@@ -107,7 +107,6 @@ rust_binary {
         "libenumn",
         "libremain",
     ],
-    vendor_available: true,
     visibility: ["//visibility:public"],
     arch: {
         x86_64: {
@@ -125,9 +124,6 @@ rust_binary {
                 "gpu",
                 "virgl_renderer_next",
             ],
-            strip: {
-                keep_symbols_and_debug_frame: true,
-            },
         },
         android: {
             shared_libs: [
@@ -138,6 +139,18 @@ rust_binary {
                 "libprocessgroup",
             ],
         },
+        linux_bionic_arm64: {
+            relative_install_path: "aarch64-linux-bionic",
+        },
+        linux_glibc_x86_64: {
+            relative_install_path: "x86_64-linux-gnu",
+        },
+        linux_musl_x86_64: {
+            relative_install_path: "x86_64-linux-musl",
+        },
+        linux_musl_arm64: {
+            relative_install_path: "aarch64-linux-musl",
+        },
     },
     ld_flags: [
         "-Wl,--rpath,\\$$ORIGIN",
@@ -146,6 +158,26 @@ rust_binary {
 
 }
 
+rust_binary {
+    name: "crosvm",
+    host_supported: true,
+    defaults: ["crosvm_binary_defaults"],
+    target: {
+        host: {
+            strip: {
+                keep_symbols_and_debug_frame: true,
+            },
+        },
+    },
+}
+
+rust_binary {
+    name: "crosvm_vendor",
+    stem: "crosvm",
+    vendor: true,
+    defaults: ["crosvm_binary_defaults"],
+}
+
 rust_test {
     name: "crosvm_test_src_main",
     defaults: ["crosvm_inner_defaults"],
