diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index f2c450cb4..177b0e352 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -14,6 +14,7 @@ rust_library {
     name: "librutabaga_gfx",
     defaults: ["crosvm_inner_defaults"],
     host_supported: true,
+    vendor_available: true,
     crate_name: "rutabaga_gfx",
     cargo_env_compat: true,
     cargo_pkg_version: "0.1.2",
@@ -29,12 +30,29 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-        "libgbm",
-        "libvirglrenderer",
+    cfgs: [
+        "gfxstream_unstable",
     ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
 
 rust_test {
@@ -61,10 +79,24 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-        "libgbm",
-        "libvirglrenderer",
-    ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
