diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index a9505616c..3ae03f1d0 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -33,12 +33,29 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-        "libgbm",
-        "libvirglrenderer",
+    cfgs: [
+        "gfxstream_unstable",
     ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
 
 rust_test {
@@ -69,10 +86,24 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-        "libgbm",
-        "libvirglrenderer",
-    ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
