diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index c69e8a386..c788314a1 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -35,14 +35,73 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
+    target: {
+        android: {
+            shared_libs: [
+                "libnativewindow",
+                "libcutils",
+                "liblog",
+            ],
+        },
+        host: {
+            features: ["vulkano"],
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+    },
     static_libs: [
+        "libdrm",
         "libgbm",
         "libvirglrenderer",
     ],
     shared_libs: [
-        "libdrm",
+        "libc++",
+        "libbase",
         "libepoxy",
+        "libgfxstream_backend",
+    ],
+}
+
+rust_library {
+    name: "librutabaga_gfx_gfxstream",
+    defaults: ["crosvm_inner_defaults"],
+    host_supported: true,
+    vendor_available: true,
+    crate_name: "rutabaga_gfx",
+    cargo_env_compat: true,
+    cargo_pkg_version: "0.1.2",
+    srcs: ["src/lib.rs"],
+    edition: "2021",
+    rustlibs: [
+        "libcfg_if",
+        "liblibc",
+        "liblog_rust",
+        "libnix",
+        "libthiserror",
+        "libzerocopy",
+    ],
+    proc_macros: ["libremain"],
+    cfgs: [
+        "gfxstream_unstable",
     ],
+    features: [
+        "gfxstream",
+    ],
+    shared_libs: [
+        "libgfxstream_backend",
+    ],
+    target: {
+        host: {
+            features: ["vulkano"],
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+    },
 }
 
 rust_test {
@@ -74,11 +133,12 @@ rust_test {
     ],
     proc_macros: ["libremain"],
     static_libs: [
+        "libdrm",
         "libgbm",
         "libvirglrenderer",
     ],
     shared_libs: [
-        "libdrm",
         "libepoxy",
+        "libgfxstream_backend",
     ],
 }
