diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index a9505616c..8dcf85f91 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -21,7 +21,6 @@ rust_library {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -37,7 +36,9 @@ rust_library {
         "libdrm",
         "libepoxy",
         "libgbm",
+        "libgfxstream_backend",
         "libvirglrenderer",
+        // "libcutils", // TODO: needed?
     ],
 }
 
@@ -57,7 +58,6 @@ rust_test {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -73,6 +73,8 @@ rust_test {
         "libdrm",
         "libepoxy",
         "libgbm",
+        "libgfxstream_backend",
         "libvirglrenderer",
+        // "libcutils", // TODO: needed?
     ],
 }
