diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index f403c0991..e901c5285 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -14,7 +14,10 @@ package {
 
 rust_library {
     name: "librutabaga_gfx",
-    defaults: ["crosvm_defaults"],
+    defaults: [
+        "crosvm_defaults",
+        "rutabaga_gfx_defaults",
+    ],
     host_supported: true,
     crate_name: "rutabaga_gfx",
     cargo_env_compat: true,
@@ -23,7 +26,6 @@ rust_library {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -37,7 +39,7 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
+    static_libs: [
         "libepoxy",
         "libvirglrenderer",
     ],
@@ -46,7 +48,10 @@ rust_library {
 
 rust_test {
     name: "rutabaga_gfx_test_src_lib",
-    defaults: ["crosvm_defaults"],
+    defaults: [
+        "crosvm_defaults",
+        "rutabaga_gfx_defaults",
+    ],
     host_supported: true,
     crate_name: "rutabaga_gfx",
     cargo_env_compat: true,
@@ -60,7 +65,6 @@ rust_test {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -75,7 +79,7 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
+    static_libs: [
         "libepoxy",
         "libvirglrenderer",
     ],
